### ISSUE TITLE
Flatpak Application TRACKING ISSUE

### ISSUE THREAD

**Issue by powpingdone** [OrcaSlicer/OrcaSlicer#3949]
This is a tracking issue for the flatpak build of orcaslicer, as there are a few specific bugs that must be fixed in order to consider the build working properly.

## Original Issue Description 

I am requesting permission to create a Flatpak application ([manifest here](https://github.com/powpingdone/flathub/tree/orcaslicer)), along with any modifications required by other contributors. Given permission, I will go on to flathub and publish the application.

**Comment by powpingdone** [comment]
@SoftFever I'm very sorry for pinging you, but I would like your "ok" on this.

**Comment by SoftFever** [comment]
Hi @powpingdone 

First thing first, that's great!


Could you please clarify what permission you are asking for? At the moment, Orca's CICD only creates appimages.
If you are working on building or GitHub actions scripts, would you mind submitting a pull request? Please let me know if you meant something else.

**Comment by powpingdone** [comment]
From https://docs.flathub.org/docs/for-app-authors/submission#theres-an-app-that-id-like-to-see-on-flathub-but-im-not-the-developer

> *If there’s an app that you'd like to be distributed on Flathub, the best first course of action is to approach the app’s developers and ask them to submit it.* Remember that for some projects or communities, this may be the first that they have heard of the Flatpak technology or interacted with somebody who advocates it. When you are making such requests, please be aware that the people you are talking to may be forming their first impressions of Flatpak as a technology and a community. 

I could do initial distribution of both nightly, and (eventually) stable releases past v2 (due to that flatpak patch needed) using CI, but since I'm not the owner of the repository I would like your permission to operate this.

**Comment by powpingdone** [comment]
I'm not asking you to submit a flatpak, I would be doing that myself (with input from you and other devs).

**Comment by SoftFever** [comment]
> From https://docs.flathub.org/docs/for-app-authors/submission#theres-an-app-that-id-like-to-see-on-flathub-but-im-not-the-developer
> 
> > **If there’s an app that you'd like to be distributed on Flathub, the best first course of action is to approach the app’s developers and ask them to submit it.** Remember that for some projects or communities, this may be the first that they have heard of the Flatpak technology or interacted with somebody who advocates it. When you are making such requests, please be aware that the people you are talking to may be forming their first impressions of Flatpak as a technology and a community.
> 
> I could do initial distribution of both nightly, and (eventually) stable releases past v2 (due to that flatpak patch needed) without using CI, but since I'm not the owner of the repository I would like your permission to operate this.

Thanks for the explanation.

Currently, all OrcaSlicer builds, including the official releases, are fully automated within this GitHub repository and it's CI/CD pipeline to ensure security. 
I intend to maintain this approach for any new distribution formats, purely from a security and maintenance perspective. 
I hope you can understand my rationale.

**Comment by powpingdone** [comment]
That's completely understandable. Shall I setup/pr a CI job to distribute a flatpak then?

**Comment by SoftFever** [comment]
> That's completely understandable. Shall I setup/pr a CI job to distribute a flatpak then? I would have to give you a key as a secret to allow uploading to flathub.

👍 
That would be great!

**Comment by ClaytonRehder** [comment]
+1. Good initiative.

**Comment by tapir** [comment]
this is great! can we not wait for 2.0 and just have one for 1.9 already?

**Comment by powpingdone** [comment]
@tapir afaict, because of the flatpak specific patches, no.

**Comment by hadess** [comment]
> I am requesting permission to create a Flatpak application ([manifest here](https://github.com/powpingdone/flathub/tree/orcaslicer)), along with any modifications required by other contributors. Given permission, I will go on to flathub and publish the application.

That manifest looks eerily familiar ;)

Let me know if you think there's anything we could do to make it easier to share between the 2 manifests repositories, once this has landed, and please CC: me on the Flathub review as well.

**Comment by powpingdone** [comment]
shhhh I totally didn't do anything to it :P

**Comment by kugelzucker** [comment]
so as it happens i just opened the orcaslicer github page to check if there was a release for 2.0 since bambustudio messed up some tree supports. 6 minutes ago was the release of 2.0, i am in luck.

since this is an open source project and devs are always very interested in annoying and pushy comments i thought i would do my part and ask:

why is there no flatpak released 6 minutes after the official release? i cant believe it.

seriously though, what is the process and how can we help? is this already implemented and the pipeline is just slower (6 minutes, i mean cmon ... :D)?

thanks for all your great work!

**Comment by powpingdone** [comment]
@kugelzucker currently i'm bogged down by irl stuff and ci is failing for reasons unknown (when it was previously fine).

**Comment by kugelzucker** [comment]
im giving this a nudge, checking in

**Comment by kugelzucker** [comment]
> im giving this a nudge, checking in

also, this builds and installs fine for me. fyi

**Comment by powpingdone** [comment]
@kugelzucker tl;dr: this messes with a lot of dep resolution stuff so it gets messy on other systems. it's fine for flatpak, but it's not fine for literally everyone else

**Comment by powpingdone** [comment]
Update for everyone who is following this but not the PR: #4175 has been merged. Next up, Softfever and I will have to get in contact with Flathub to get the necessary things (api keys) to upload from CI.

**Comment by powpingdone** [comment]
cc #5449

**Comment by powpingdone** [comment]
cc #5474

**Comment by cochcoder** [comment]
I'd also like to note that you may want to build the flatpak with freedesktop-22.08 rather than gnome-45, as freedesktop prioritizes having applications work consistently on all platforms rather than one. I also noticed in my testing that building with freedesktop causes the flatpak to have a slightly smaller size and uses less ram than the gnome build.

**Comment by powpingdone** [comment]
@hgelpke cross quoting from pr:
> Can somebody give a rundown on how to build this? Tried on Fedora 40 and failed but it is also the first time I'm trying this.
tldr: `flatpak-builder --force-clean --user --install --jobs=2 builddir io.github.softfever.OrcaSlicer.yml`. 
Adjust `--jobs=` to fit in RAM, as a single compile job in OrcaSlicer can take up 4GB ram or more.
Build is currently broken and i'm investigating in #5510.

**Comment by hgelpke** [comment]
> @hgelpke cross quoting from pr:
> 
> > Can somebody give a rundown on how to build this? Tried on Fedora 40 and failed but it is also the first time I'm trying this.
> 
> tldr: `flatpak-builder --force-clean --user --install --jobs=2 builddir io.github.softfever.OrcaSlicer.yml`. Adjust `--jobs=` to fit in RAM, as a single compile job in OrcaSlicer can take up 4GB ram or more. Build is currently broken and i'm investigating in #5510.

Great! Thanks for the reply and apologies for missing it. Good luck on the fix and I appreciate your efforts

**Comment by powpingdone** [comment]
Ok, after the cmake fiasco we're now in the tree. Finally. Going to file an issue to flathub soon(tm).

**Comment by albei** [comment]
P1S camera still not working Flatpak 2.1.1 Build all #2600
Testing it on a fedora silverblue.
"Player is malfunctioning. Please reinstall the system player. [2:2]"
Even so we finally have a flatpak! Many thanks for your effort !

**Comment by powpingdone** [comment]
@albei tracked in a separate issue #5449, iirc that's a external patch currently. I do want to do the gstreamer stuff properly.

**Comment by powpingdone** [comment]
https://discourse.flathub.org/t/orcaslicer-requesting-external-upload-to-flathub/7340 @SoftFever (also @hadess because I don't know where to go with this situation).

**Comment by tapir** [comment]
Does this have to be fully automated through CI @SoftFever? What if flathub doesn't want to do an external upload?

**Comment by JustinLex** [comment]
Is there an issue with letting Flathub build the flatpak themselves? It's important for security reasons that Flathub is able to build from source, and the build process and configuration is identical to what is already in the Github Actions CI. Flatpak builds can continue to run in Github Actions CI as an internal testing step, but it looks like the build distributed by Flathub should also be built by Flathub.

**Comment by tapir** [comment]
Looks like the issue on flathub side was closed due to inactivity. I'd like to take this over if possible but I, 100%, agree with @JustinLex, I don't see any reason to not follow flathub standards. Are we missing something @powpingdone?

**Comment by powpingdone** [comment]
@tapir no it's just been me being incredibly busy. I'll make the flathub issue proper (not discourse).

**Comment by powpingdone** [comment]
I misread your comment, sorry. @SoftFever why can't we have flathub building the flatpak for orcaslicer externally?

**Comment by powpingdone** [comment]
This would make distribution a lot easier because you can just submodule the repo to them and still have CI here.

**Comment by powpingdone** [comment]
@SoftFever bump, waiting for your input.

**Comment by Ziomal12** [comment]
Has anyone got working flatpak of 2.2?

**Comment by cochcoder** [comment]
@powpingdone 

I don't mean to be that person, but do you know when we could get a working GitHub action for the flatpak?

**Comment by anarsoul** [comment]
@SoftFever sorry for nagging you, but could we please get an approval for building flatpak externally?

It would simplify installation of Orca Slicer for a lot of Linux users and personally I don't see any drawbacks for you, only benefits.

Flatpak runtime environment is more contained than any Linux distro, so Orca Slicer will be running more stable and it would be easier to reproduce bugs if there are any.

**Comment by Ziomal12** [comment]
@anarsoul Correct me if I'm wrong but as far as I understand under AGPL-3.0 licence distribution of the software is allowed as you disclose source and use the same licence as original.

**Comment by anarsoul** [comment]
@Ziomal12 see flathub submission guide: https://docs.flathub.org/docs/for-app-authors/submission/#theres-an-app-that-id-like-to-see-on-flathub-but-im-not-the-developer

We are at the step one. Submitting it without author's approval is the last resort.

**Comment by cochcoder** [comment]
@ChaosBlades 
It's only been 12 hours, give SoftFever some time. Both SoftFever and PowPingDone are really busy.

**Comment by tapir** [comment]
@cochcoder we haven't heard from him on this thread since Aug. That's probably what @ChaosBlades meant.

**Comment by anarsoul** [comment]
I created a repo with (fixed) flatpak manifest for OrcaSlicer: https://github.com/anarsoul/io.github.softfever.OrcaSlicer

Flatpak for v2.2.0-rc can be found here: https://github.com/anarsoul/io.github.softfever.OrcaSlicer/releases/tag/v2.2.0-rc

**Comment by cochcoder** [comment]
@anarsoul 

Do you think you'll be able to make a PR with the manifest fixes?

**Comment by anarsoul** [comment]
@cochcoder I can, although it won't help with getting flatpak into flathub.

One of the fixes has been sitting in review for 3 weeks: https://github.com/SoftFever/OrcaSlicer/pull/7057

The other fix is getting rid of python script and integrating dark theme detection into wxWidgets (see patch in my repo, I took it from bambulab slicer manifest repo)

**Comment by cochcoder** [comment]
> @cochcoder I can, although it won't help with getting flatpak into flathub.
> 

That's fine, we would've need to get a working flatpak first anyways. 

> 
> One of the fixes has been sitting in review for 3 weeks: https://github.com/SoftFever/OrcaSlicer/pull/7057
> 
> 
> 
> The other fix is getting rid of python script and integrating dark theme detection into wxWidgets (see the patch in my repo, I took it from bambulab slicer manifest repo)
> 
> 

You could link to those in your PR and mention that they need to get merged alongside the manifest fixes.

**Comment by anarsoul** [comment]
> If the OrcaSlicer devs are anything like me, they're probably expecting better commit messages.

It's not my PR, but if OrcaSlicer devs were anything like me, they'd probably fix up commit message if they don't like it and merge the PR :)

**Comment by trebory6** [comment]
> @ChaosBlades It's only been 12 hours, give SoftFever some time. Both SoftFever and PowPingDone are really busy.

It's been 3 weeks, is that enough time? haha

Seriously though, it's disappointing that flatpak compatibility is taking so long. I'm getting very tired of Orcaslicer breaking with each new Kernel update.

**Comment by anarsoul** [comment]
> Seriously though, it's disappointing that flatpak compatibility is taking so long officially. I'm getting very tired of bandaging Orcaslicer's appimage with environment variables after it breaks with each new Kernel update.

Just install flatpak from https://github.com/anarsoul/io.github.softfever.OrcaSlicer/releases/tag/v2.2.0-1

**Comment by anarsoul** [comment]
@SoftFever sorry to nag you again, but can we get this issue moving? If you need to discuss it further in real-time, text or voice, I'm available in discord on Orca Slicer server, same nickname as here.

**Comment by Skaifer** [comment]
Feels so right to install a flatpak. I have issue though with flatpak version as well. The preview side is all gray. Terminal output - https://paste.pika-os.com/p/sloth-camel-ape. Tried to set env variables as for appimage, but that doesn't do anything or makes things worse. And appimage currently runs with those fixes, but preview is terribly slow.

**Comment by Jookia** [comment]
As a user I would really appreciate an official Flathub package even if it's not created by the OrcaSlicer devs.

Long term this might be the more secure option too: Right now to run OrcaSlicer I have to download binaries with no way or ability to check if they're built from corresponding source code. I can't even check if the binaries are the ones the developer uploaded, there's no package signing.

Having an independent third party such as Flathub build and distribute binaries based on a manifest would give me an ability to trust the version of OrcaSlicer I'm running.

**Comment by Skaifer** [comment]
Ran Flatpak version successfully on Bazzite under Wayland, thanks to [Fr0sty4 comment](https://github.com/SoftFever/OrcaSlicer/issues/6900#issuecomment-2480244457):
> For anyone using the flatpak -- adding this environment variable via Flatseal should work for software rendering and stays inside the flatpak:

`__EGL_VENDOR_LIBRARY_FILENAMES=/usr/lib/x86_64-linux-gnu/GL/default/share/glvnd/egl_vendor.d/50_mesa.json`

The thing, that's different from previous variable, is the path to 50_mesa.json, which previously was `/usr/share/glvnd/egl_vendor.d/50_mesa.json`. I checked the new path, and there is nothing. Basically giving it nothing fixed the issue, which is weird, but I digress. The viewport when viewing the sliced part is slow, but that's what we get until it gets fixed for Nvidia users.

**Comment by anarsoul** [comment]
> until it gets fixed for Nvidia users

Someone needs to report it to NVIDIA, it's very likely to be their bug

**Comment by Skaifer** [comment]
[MackBambu](https://github.com/bambulab/BambuStudio/issues/4669#issuecomment-2495929679) at BambuStudio thread says Nvidia has dropped old OpenGL version support, so they are now working on a fix, which hopefully would eventually come to OrcaSlicer and heal our sores.

**Comment by egruttum** [comment]
> I created a repo with (fixed) flatpak manifest for OrcaSlicer: https://github.com/anarsoul/io.github.softfever.OrcaSlicer
> 
> Flatpak for v2.2.0 can be found here: https://github.com/anarsoul/io.github.softfever.OrcaSlicer/releases/tag/v2.2.0-1

Thank you for this.  I drove myself crazy trying to make OrcaSlicer work on my Chromebook with the AppImage.  This worked so easily.  Hopefully this gets into the official Flatpak repository someday to make it even easier for people to find.

**Comment by Prazgreen1** [comment]
> > I created a repo with (fixed) flatpak manifest for OrcaSlicer: https://github.com/anarsoul/io.github.softfever.OrcaSlicer
> > Flatpak for v2.2.0 can be found here: https://github.com/anarsoul/io.github.softfever.OrcaSlicer/releases/tag/v2.2.0-1
> 
> Thank you for this. I drove myself crazy trying to make OrcaSlicer work on my Chromebook with the AppImage. This worked so easily. Hopefully this gets into the official Flatpak repository someday to make it even easier for people to find.

this is in the official repo

**Comment by mtorromeo** [comment]
> this is in the official repo

Which official repo are you talking about?

It's not here https://flathub.org/apps/search?q=orcaslicer

**Comment by 616b2f** [comment]
For those who are not aware there is already a Flatpak file in the release bundle it's easy to install, just download and open it with you Software Store application (it's discovered by the most distros automatically, when you try to open an XY.flatpak file.

I hope this helps someone, as I was also searching for a way to use the Flatpak version of Orca.

**Comment by CharlemagneLasse** [comment]
> For those who are not aware there is already a Flatpak file in the release bundle it's easy to install, just download and open it with you Software Store application (it's discovered by the most distros automatically, when you try to open an XY.flatpak file).
> 
> I hope this helps someone, as I was also searching for a way to use the Flatpak version of Orca.

Such a flatpak is only really useful when it is auto-updating and can actually be found by the community using the builtin flatpak repo management tools.

And https://github.com/anarsoul/io.github.softfever.OrcaSlicer/ seems to be the only one which seems to work. It might therefore be good to use this one on flathub.

The attempt to do this was unfortunately blocked by @SoftFever (for unknown reasons because no feedback is given by @SoftFever)

**Comment by Jookia** [comment]
I believe the reasoning is that SoftFever doesn't want people to package it externally, so someone would have to make a fork and rebrand it for Flathub.

**Comment by CharlemagneLasse** [comment]
> I believe the reasoning is that SoftFever doesn't want people to package it externally, so someone would have to make a fork and rebrand it for Flathub.

But then @SoftFever should upload it to flathub and accept fixes from others

**Comment by Jookia** [comment]
Getting a manifest on Flathub means either:

- Flathub does the build, not GitHub CI (SoftFever doesn't want this)
- SoftFever negotiates to have direct push access to Flathub

Looking at the thread here, SoftFever doesn't seem interested in the
second and has denied doing the first.

I don't know if Flathub lets people package software without the
author's permission, but I imagine SoftFever would not be happy with
having unauthorized 'OrcaSlicer' builds floating around distribution
channels.

A third option is to host a Flatpak repo and distribute a ref for it.
Then SoftFever can bypass all Flathub policy and ship the application
directly to users. This would be great for updates.

Alternatively someone else could create their own Flatpak repository and
ship OrcaSlicer in it. This once again hits the 'someone is distributing
unofficial builds'.

If the community wants a Flathub build, it will probably have to be done
using a fork. This isn't great security wise, but at least nobody would
be unhappy and nobody would be fooled about unofficial applications.

**Comment by queeup** [comment]
2000+ desktop apps are doing bad about distributing theirs apps from flathub? I don't get it... What is the problem about distributing this app from flathub for Linux users?

https://flathub.org/statistics

**Comment by eternal-sorrow** [comment]
>I believe the reasoning is that SoftFever doesn't want people

>(SoftFever doesn't want this)

You believe or you know this? And why @SoftFever doesn't speak for oneself?

**Comment by rehashedsalt** [comment]
> I believe the reasoning is that SoftFever doesn't want people to package it externally, so someone would have to make a fork and rebrand it for Flathub.

The issue remains open and attached to a GH Project. If the maintainers were that deadset against a FlatHub build, neither of these would be true. Please don't speculate.

**Comment by Jookia** [comment]
Alright sorry, comment deleted.

**Comment by ClaytonRehder** [comment]
Thank you all for creating the flatpak version, it is working perfectly. I would like to know if/when there will be a version available on flathub later on? are there plans for that?

**Comment by ChaosBlades** [comment]
Are we still waiting on SoftFever approval for flathub or is there some other blocker?

**Comment by baduhai** [comment]
> Are we still waiting on SoftFever approval for flathub or is there some other blocker?

Approval is all that's missing

**Comment by joepalma** [comment]
@SoftFever any progress on this? Having it in the repo is the whole point of the flatpak, ease of installation and upgrades. Without a repo, as of right now, there's effectively no difference between the flatpak and appimage to the end user. Both are equally annoying to deal with.

**Comment by eternal-sorrow** [comment]
Not necessarily publish it on flathub, may host you own repo if you want to. This is the advantage of Flatpak compared to Snap for example.

**Comment by eternal-sorrow** [comment]
Not stale, automated issue closing shouldn't be a thing.

**Comment by Prazgreen1** [comment]
Comment to prevent stale
Approval for flsthub still waiting on?

**Comment by razzeee** [comment]
This should be the necessary repo setup, that can be submitted to flathub.

https://github.com/razzeee/io.github.softfever.OrcaSlicer

There are two issues, the linter surfaces, that will need addressing, when it's submitted, the `home` permission and the talk name.

**Comment by hadess** [comment]
> There are two issues, the linter surfaces, that will need addressing, when it's submitted, the `home` permission and the talk name.

See this for the talk name:
https://github.com/flathub/com.bambulab.BambuStudio/commit/9b3585de8e037665a5473a5b12c89421b7e2dbd5

Feel free to copy/paste the explanation.

**Comment by razzeee** [comment]
Pushed thank you

**Comment by SoftFever** [comment]
I know this has taken a long time to address, and I appreciate your patience. I’m sorry about the delay too.
Over the past year I’ve been dealing with a difficult situation in my personal life (unrelated to the project), which slowed things down on my side.

@powpingdone Please let me know if you’re still interested.
If you’d like, feel free to ping me on Discord or reply here.

I’d also appreciate help from anyone with Flatpak/Flathub experience. If you’re available, please feel free to chime in or ping me — I’d be grateful 🙏

**Comment by razzeee** [comment]
As said, you should be fine checking and commiting https://github.com/razzeee/io.github.softfever.OrcaSlicer?rgh-link-date=2025-10-06T01%3A05%3A20.000Z

unless flathub moved the goal post since then

**Comment by SoftFever** [comment]
> As said, you should be fine checking and commiting https://github.com/razzeee/io.github.softfever.OrcaSlicer?rgh-link-date=2025-10-06T01%3A05%3A20.000Z
> 
> unless flathub moved the goal post since then

Thanks!
How do we manage official/alpha/beta/nightly releases on Flathub? Any suggestions warnings before I dig into the rabbit hole? ;)

**Comment by hadess** [comment]
> How do we manage official/alpha/beta/nightly releases on Flathub? Any suggestions warnings before I dig into the rabbit hole? ;)

Nightlies and other CI/CD builds should not go on Flathub, they don't have the build capacity to handle that amount of traffic. For betas, and regular development builds, you can ask for a separate repo, usually `$APP.Devel`, which will be installable from the `flathub-beta` repository, and won't appear on Flathub.org.

Note that this also requires your app to build as a a separate app-id, probably with a different icon, etc.

Not the most well-maintained example, but this is what I did:
- https://github.com/flathub/org.gnome.Totem.Devel/
- https://github.com/flathub/org.gnome.Totem

<img width="396" height="247" alt="Image" src="https://github.com/user-attachments/assets/d8d33c69-b6a2-4894-823e-a9970812eb52" />

**Comment by razzeee** [comment]
You should read https://docs.flathub.org/docs/for-app-authors/submission and the rest of the docs, it should cover all things.

In the end, you will have a repo over at github/flathub, where you can update the build files.

**Comment by SoftFever** [comment]
> > How do we manage official/alpha/beta/nightly releases on Flathub? Any suggestions warnings before I dig into the rabbit hole? ;)
> 
> Nightlies and other CI/CD builds should not go on Flathub, they don't have the build capacity to handle that amount of traffic. For betas, and regular development builds, you can ask for a separate repo, usually `$APP.Devel`, which will be installable from the `flathub-beta` repository, and won't appear on Flathub.org.
> 
> Note that this also requires your app to build as a a separate app-id, probably with a different icon, etc.
> 
> Not the most well-maintained example, but this is what I did:
> - https://github.com/flathub/org.gnome.Totem.Devel/
> - https://github.com/flathub/org.gnome.Totem
> 
> <img width="396" height="247" alt="Image" src="https://github.com/user-attachments/assets/d8d33c69-b6a2-4894-823e-a9970812eb52" />

Thank you for sharing your insights. This makes perfect sense. It’s great to hear from someone with firsthand experience!

**Comment by eternal-sorrow** [comment]
>or betas, and regular development builds, you can ask for a separate repo, usually $APP.Devel

I don't see why a separate repo is needed. Don't most of the apps just use a `beta` branch in the main repo?

**Comment by razzeee** [comment]
> > or betas, and regular development builds, you can ask for a separate repo, usually $APP.Devel
> 
> I don't see why a separate repo is needed. Don't most of the apps just use a `beta` branch in the main repo?

Yes, there are two approaches, both have their ups and downs

**Comment by hadess** [comment]
> I don't see why a separate repo is needed. Don't most of the apps just use a `beta` branch in the main repo?

Depends on how you set things up, it's expected that the `beta` branch to actually be a pre-release of what's going to end up in the main branch in short order, and because the app-id is shared, users don't really control which one is going to be launched when using the app icon.

Having a separate app-id and icon means that users could always use the latest until it breaks, and they still have easy access to the stable version. Using `beta` branch is more of a faff and command-line.

**Comment by coryrc** [comment]
Appimage build uses the binaries built for Ubuntu, which are going to be built regardless. Do we have to waste CPU time and drive space also building a duplicate of those binaries for Flatpak instead of packaging the same binaries?

**Comment by razzeee** [comment]
> Appimage build uses the binaries built for Ubuntu, which are going to be built regardless. Do we have to waste CPU time and drive space also building a duplicate of those binaries for Flatpak instead of packaging the same binaries?

You can repackage those binaries, but it's usually discouraged, as people then don't ship every architecture. They usually don't care to collect binaries for arches like aarch64, which you would get for free, when building natively.

**Comment by coryrc** [comment]
> You can repackage those binaries, but it's usually discouraged, as people then don't ship every architecture. They usually don't care to collect binaries for arches like aarch64, which you would get for free, when building natively.

Can you build just aarch64 and reuse the x64 build?

Context is cleaning up build workflows and [cache being mostly occupied by flatpak entries](https://github.com/OrcaSlicer/OrcaSlicer/issues/11602#issuecomment-3629809612) which could trivially be disabled, but I'd rather shrink it down to a reasonable size and keep the cache, if it's doing anything. Not having a second build of x64 should shrink the cache in half right off the bat?

**Comment by razzeee** [comment]
Will you host your own remote? Otherwise it should be build by flathub anyway. Or are we talking beta/nightly builds?

**Comment by anarsoul** [comment]
> Appimage build uses the binaries built for Ubuntu, which are going to be built regardless. Do we have to waste CPU time and drive space also building a duplicate of those binaries for Flatpak instead of packaging the same binaries?

Please don't. Flatpak is currently built against Flatpak runtime, while Appimage e.g. requires webkit-gtk from Ubuntu and doesn't work on majority of other distros.

It would make more sense to abandon appimage build in favor of flatpak.

**Comment by coryrc** [comment]
> Or are we talking beta/nightly builds?

Right, the builds that happen on Github Actions on every single PR push.

>  Flatpak is currently built against Flatpak runtime, while Appimage e.g. requires webkit-gtk from Ubuntu and doesn't work on majority of other distros.

You're wrong, AppImage works on many different kinds of distros, as reported in #10990; the problem being people want to run old distros with new Appimage. You can't support everything forever.

> It would make more sense to abandon appimage build in favor of flatpak.

How would I extract the unit tests and run them from the flatpak?

**Comment by anarsoul** [comment]
> the problem being people want to run old distros with new Appimage. You can't support everything forever.

It doesn't work on Arch or Debian. It literally targets whatever is present in Ubuntu.

> You can't support everything forever.

That's where Flatpak helps. You have predictable environment anywhere where Flatpak is supported. Multiple versions of runtime are supported.

> How would I extract the unit tests and run them from the flatpak?

No idea, you should probably ask upstream.

**Comment by coryrc** [comment]
> > the problem being people want to run old distros with new Appimage. You can't support everything forever.
> 
> It doesn't work on Arch or Debian. It literally targets whatever is present in Ubuntu.

Wrong again, [here it's reported as working on Arch](https://github.com/OrcaSlicer/OrcaSlicer/issues/10990#issuecomment-3389946845). And the whole reason for the Issue is that it did work on lots of distros and people thought only Ubuntu is now supported due to some naming.

> > You can't support everything forever.
> 
> That's where Flatpak helps. You have predictable environment anywhere where Flatpak is supported. Multiple versions of runtime are supported.

That very factor of bundling the runtime has spawned a number of issues: #6433 #8145 #8720 #10815 #10034 

You've just pushed the burden of occasionally updating your own system onto the project maintainers. If you want to run a system from years ago, then run a slicer from years ago.

**Comment by anarsoul** [comment]
> Wrong again, https://github.com/OrcaSlicer/OrcaSlicer/issues/10990#issuecomment-3389946845. And the whole reason for the Issue is that it did work on lots of distros and people thought only Ubuntu is now supported due to some naming.

It's only a matter when webkit-gtk2 gets updated in Arch and Ubuntu stays behind once again

> You've just pushed the burden of occasionally updating your own system onto the project maintainers. If you want to run a system from years ago, then run a slicer from years ago.

It's quite the opposite. Ubuntu usually ships outdated libraries

But anyway, I see Flatpak as universal solution that works for everyone, while Appimage is Ubuntu-specific (built against the libraries shipped with Ubuntu). If you want to drop one, the obvious choice is Appimage, not Flatpak.

**Comment by anarsoul** [comment]
> That very factor of bundling the runtime has spawned a number of issues

Most of these are caused by NVIDIA driver not supporting legacy profile for GL, right? It isn't exactly Flatpak issue.

**Comment by razzeee** [comment]
> That very factor of bundling the runtime has spawned a number of issues: [#6433](https://github.com/OrcaSlicer/OrcaSlicer/issues/6433) [#8145](https://github.com/OrcaSlicer/OrcaSlicer/issues/8145) [#8720](https://github.com/OrcaSlicer/OrcaSlicer/issues/8720) [#10815](https://github.com/OrcaSlicer/OrcaSlicer/issues/10815) [#10034](https://github.com/OrcaSlicer/OrcaSlicer/issues/10034)


All of these issues claim that the problems also exist with the appimage. Only the dark color one is specific, looking at the comments.

**Comment by coryrc** [comment]
> > That very factor of bundling the runtime has spawned a number of issues: [#6433](https://github.com/OrcaSlicer/OrcaSlicer/issues/6433) [#8145](https://github.com/OrcaSlicer/OrcaSlicer/issues/8145) [#8720](https://github.com/OrcaSlicer/OrcaSlicer/issues/8720) [#10815](https://github.com/OrcaSlicer/OrcaSlicer/issues/10815) [#10034](https://github.com/OrcaSlicer/OrcaSlicer/issues/10034)
> 
> All of these issues claim that the problems also exist with the appimage. Only the dark color one is specific, looking at the comments.

I was wrong on 6433.

8145: [Eitherway, if I download the appimage version it does work fine, so there's clearly some underlying difference between the two that makes the flatpak simply not like Nvidia for some reason.](https://github.com/OrcaSlicer/OrcaSlicer/issues/8145#issuecomment-3649053645)

8720: one person said neither worked for them, everyone else was specific to flatpak

10815: [using the appimage version seems to work fine (it includes a workaround to use zink on NVIDIA to translate OpenGL calls to Vulkan)](https://github.com/OrcaSlicer/OrcaSlicer/issues/10815#issuecomment-3548195691)

**Comment by coryrc** [comment]
> But anyway, I see Flatpak as universal solution that works for everyone, while Appimage is Ubuntu-specific (built against the libraries shipped with Ubuntu). If you want to drop one, the obvious choice is Appimage, not Flatpak.

Windows: cmake --build, run binary
Mac: cmake --build, run binary
Linux: cmake --build, run binary
Flatpak: piles of yaml create containerized thing I can't(?) run outside the container

The Linux build compiles on Arch, Fedora, Mint, Debian, Ubuntu, and most other things based on those; and the Ubuntu 20 AppImage was/is widely working and is a simple wrapper around the binary.

Only one of them is the odd-man-out and requires specialized handling.

**Comment by eternal-sorrow** [comment]
>containerized thing I can't(?) run outside the container

Good.

**Comment by coryrc** [comment]
Look, if it's infeasible to use the compiled x64 binary in flatpak, so be it. But if it's merely "distasteful" and actually does work (because, after all, flatpak is basically its own distro running in containers, and either the built binary works in the flatpak or it doesn't), then lets save a lot of time and cache wasted in the PRs on binaries nobody and nothing usually runs (flatpak) and copy from the one actually necessary to be built to run the tests ("Linux"). For releases or "nightly" builds, they can still do a full "flatpak" build.

My issue is Flatpak occupies a lot of disk and cache quota from PR builds and very rarely is this build actually needed (AFAICT most users are on Windows), but I expect rarely a Flatpak user will want to download a PR build so it's highly inconvenient to not have them.

**Comment by anarsoul** [comment]
It isn't "merely distasteful", you are building a binary that targets a different environment from flatpak runtime.

Anyway, I'm not going to waste my time discussing it here anymore, I expressed my opinion and provided the reasoning, so do whatever you want.

I can always build a flatpak for myself in my own repo and share it with others.

**Comment by CharlemagneLasse** [comment]
> Look, if it's infeasible to use the compiled x64 binary in flatpak, so be it. But if it's merely "distasteful" and actually does work (because, after all, flatpak is basically its own distro running in containers, and either the built binary works in the flatpak or it doesn't), then lets save a lot of time and cache wasted in the PRs on binaries nobody and nothing usually runs (flatpak) and copy from the one actually necessary to be built to run the tests ("Linux").

What are you discussing here? [Flathub wants to rebuild the releases](https://docs.flathub.org/docs/for-app-authors/requirements#building-from-source) and [people asked for the permission to submit it to flathub](https://docs.flathub.org/docs/for-app-authors/submission#theres-an-app-that-id-like-to-see-on-flathub-but-im-not-the-developer) (as something like `io.github.OrcaSlicer.OrcaSlicer`). But due to various reason, this was never approved or rejected.

Other projects simply discussed this and then assigned a group of interested people to handle the submission of the releases to flathub. This is then handled in a separate repository under https://github.com/flathub/ and has nothing to do with the CI for your main-repo PRs

> For releases or "nightly" builds, they can still do a full "flatpak" build.

Who are "they"? Flathub? Flathub would like to [only provide stable releases](https://docs.flathub.org/docs/for-app-authors/requirements#stable-releases). They are not interested in your nightly or PR builds. This is your own cup of tea - but not the one this ticket is about.

> My issue is Flatpak occupies a lot of disk and cache quota from PR builds and very rarely is this build actually needed (AFAICT most users are on Windows), 

I don't know a single person which uses OrcaSlicer under Windows. But also only know persons which use an actual release of OrcaSlicer - but all of them would like to download new releases "automatically" via Flathub. And of course, there people which use it under Windows or which want to use only a non-Flathub flatpak. But we are currently not talking about them here.

And I would say, this comment is irrelevant for the topic of this ticket because you are talking about PRs and this ticket is about the Flathub submission. Just to have a good and reliable (and often automatically activated) flatpak source for OrcaSlicer - which is a proper source with all dependencies and automatic updates (from a user perspective).

**Comment by coryrc** [comment]
>> For releases or "nightly" builds, they can still do a full "flatpak" build.

> Who are "they"?

`releases or "nightly" builds`

> And I would say, this comment is irrelevant for the topic of this ticket because you are talking about PRs and this ticket is about the Flathub submission.

The first post is just about having a flatpak build:

> This is a tracking issue for the flatpak build of orcaslicer, as there are a few specific bugs that must be fixed in order to consider the build working properly.

And the immediate ensuing discussion includes:

> Nightlies and other CI/CD builds should not go on Flathub, they don't have the build capacity to handle that amount of traffic.

And the PR which added just the build into this repo references this issue: 

https://github.com/OrcaSlicer/OrcaSlicer/pull/4175#issue-2145552817

It's entirely your misconception that "this ticket is about the Flathub submission".

**Comment by CharlemagneLasse** [comment]
> It's entirely your misconception that "this ticket is about the Flathub submission".

I understand. This is now just about alienating people and not to get something solved.

**Comment by apparle** [comment]
In the interest of keeping the conversation constructive and letting cooler heads prevail -- there's few different things being discussed on this thread:
1. A lot of users wanting an official flatpak release & updates directly via flathub.
2. Handling of beta / nightly / PR builds for flatpak
3. The flatpak build process, as existing on this repository, being sub-optimal.

I want to highlight that a LOT of users (including myself) are focused on point-1 in this list, and aren't the primary consumers of beta/nightly/PR builds. And as @anarsoul, @razzeee and @CharlemagneLasse have been trying to highlight, _for the stable releases on flathub_, no build steps need to happen on this repo at all -- instead whole build process is done by flathub (and their CPU & disk resources) based on the manifest file in the distribution repo (typically not this repo). This repo is NOT expected to provide a built flatpak file at all.

Regarding point-2, some CI/CD build flow for this repo needs to exist so that flatpak build process isn't broken. And as I understand the current mechanism is to build a full flatpak file. @coryrc is rightfully concerned about the current build-flow for flatpak being wasteful on resource. And then the discussion digressed into a potential solution of sharing binaries between appimages and flatpak to optimize point-3. 
Taking a step back, the first question to ask -- do we even need a flatpak version for every nightly or a PR build ?
- Personally, I'd argue that this is not strictly necessary; as long as some form of Linux build works (say binaries or appimage) then that's sufficient CI coverage. And ensuring flatpak can be successfully built can instead be a final release-checklist item or maybe a beta/release-candidate item (that happens relatively infrequently). 
- Alternatively, as @anarsoul suggested, another option would be to only do flatpak build on nightly/PR builds and defer other forms of Linux builds like appimage to a final release-checklist item. 
- There's only marginal value-add in doing multiple Linux builds for every PR / nightly. But that's just my opinion and others should chime-in.

It's fair position that we need come to a reasonable solution for point-2 and point-3 as amenable to this project and the available build resources. BUT that shouldn't prevent us from making progress on point-1 above - it is `flathub builds binary from source @ specific commit-hash of this repo` completely independent of the flatpak build resources on this repo. 

@coryrc if you prefer to track all the 3 points in this issue, that's fine. But can we look at point-1 and point-2 + point-3 independently and make progress on them ?

**Comment by Alaric323** [comment]
Hi all, there appears to be an issue with 2.3.2-rc2 where the Flatpak has a different identifier from what previous versions have.  It's change from io.github.softfever.OrcaSlicer to io.github.orcaslicer.Orcaslicer.  This change breaks updates makes the flatpak install alongside the old version, not replace it.

<img width="781" height="92" alt="Image" src="https://github.com/user-attachments/assets/d1c88185-8630-4450-8e3f-33877358b11d" />

**Comment by ClaytonRehder** [comment]
Hi, any news about Flathub implementation?

**Comment by gioan777** [comment]
> Hi all, there appears to be an issue with 2.3.2-rc2 where the Flatpak has a different identifier from what previous versions have. It's change from io.github.softfever.OrcaSlicer to io.github.orcaslicer.Orcaslicer. This change breaks updates makes the flatpak install alongside the old version, not replace it.

I had the same issue, and it was annoying mainly because tons of settings I had in the old version didn't carry over and I had to setup orcaslicer from scratch. Fortunately it turns out just copying old orcaslicer config files in ~/.var/app/io.github.softfever.OrcaSlicer to the new one ~/.var/app/io.github.orcaslicer.Orcaslicer, worked fine for me. No issues with the manual migration and then I just uninstalled the old flatpak with the old identifier.

**Comment by ovitters** [comment]
> Hi, any news about Flathub implementation?

From https://github.com/OrcaSlicer/OrcaSlicer/releases/tag/v2.3.2:
> For Linux users: The official Flathub version will be available soon.

I'm quite looking forward to that (I'm not a dev, just a user)!

**Comment by Jookia** [comment]
Got hit by the Flatpak update removing all settings. Run this to restore them:

```
rsync -avz ~/.var/app/io.github.softfever.OrcaSlicer/ ~/.var/app/com.orcaslicer.OrcaSlicer/
```

**Comment by nevergone** [comment]
Is there any chance that Orca Slicer will be uploaded to Flathub?

**Comment by Dampfwalze** [comment]
> From https://github.com/OrcaSlicer/OrcaSlicer/releases/tag/v2.3.2:
> 
> > For Linux users: The official Flathub version will be available soon.

It is crossed out now?

<img width="523" height="55" alt="Image" src="https://github.com/user-attachments/assets/8e42b74d-ac74-4401-844b-8f11b5b7c36a" />

Does that mean, it is literally coming out right now, or it is postponed??

**Comment by gioan777** [comment]
It seems the devs are attempting to release it on Flathub, but there are some issues that need to be resolved.
https://github.com/flathub/flathub/pull/8181

**Comment by razzeee** [comment]
It's done

**Comment by eternal-sorrow** [comment]
https://flathub.org/en/apps/com.orcaslicer.OrcaSlicer

Already available on Flathub. Don't forget to transfer your data from the old app_id to the new one. I just did `mv ~/.var/app/{io.github.softfever.OrcaSlicer,com.orcaslicer.OrcaSlicer}`.

**Comment by Alaric323** [comment]
Downloaded to test; had to copy the data as expected.  Was able to slice a part and send to printer without issue!

Congrats Orca team on the Flathub release!

**Comment by gioan777** [comment]
It crashes on me on startup, both the flathub version and the one downloaded from the github releases v2.3.2 stable. First it complains Gnome Sdk (version 49 if downloaded from Github, version 50 if downloaded from flathub) isn't installed, and after manually installing the runtime the crash backtrace is somewhere within libstdc++. I run Arch Linux latest updates btw.

I'm too busy to properly debug it right now. Unless someone else has issues I'll assume something's wrong with my setup. For the time being I'll use v2.3.2-rc2's flatpak which works for me.

**Comment by ClaytonRehder** [comment]
Nice, that is really good news!!! Thank you very much to all the involved in this Flathub release!! Tried here and worked perfectly.

**Comment by kinghat** [comment]
dark mode doesnt appear to be working for me:

> Operating System: Fedora Linux 43
> KDE Plasma Version: 6.6.3
> KDE Frameworks Version: 6.24.0
> Qt Version: 6.10.2
> Kernel Version: 6.19.9-200.fc43.x86_64 (64-bit)
> Graphics Platform: Wayland
> Processors: 4 × Intel® Core™ i7-5600U CPU @ 2.60GHz
> Memory: 12 GiB of RAM (11.5 GiB usable)
> Graphics Processor: Intel® HD Graphics 5500

**Comment by anarsoul** [comment]
> dark mode doesnt appear to be working for me:
> 
> > Operating System: Fedora Linux 43
> > KDE Plasma Version: 6.6.3

It is an issue with flatpak and Breeze theme for Gtk. See https://github.com/flathub/org.gtk.Gtk3theme.Breeze/issues/170

As a workaround, change Gtk theme in Plasma settings to Adwaita or Adapta.

The issue is not specific to Orca Slicer, e.g. FreeCAD and KiCAD are also affected.

**Comment by piit79** [comment]
Great work, thanks!

For anyone migrating from the AppImage to Flatpak, you can run this to copy the settings (after first startup):

```
rsync -av --delete ~/.config/OrcaSlicer/ ~/.var/app/com.orcaslicer.OrcaSlicer/config/OrcaSlicer/
```

**Comment by kinghat** [comment]
> As a workaround, change Gtk theme in Plasma settings to Adwaita or Adapta.
> 
> The issue is not specific to Orca Slicer, e.g. FreeCAD and KiCAD are also affected.

Can i set this per app vs globally in plasma?

**Comment by Dampfwalze** [comment]
> Can i set this per app vs globally in plasma?

If you're issue is that it can be enabled at all, this environment variable worked for me:

```
GTK_THEME=Adwaita:dark
```

In KDE you can just add it to the applications launch configuration.

**Comment by kinghat** [comment]
> > Can i set this per app vs globally in plasma?
> 
> If you're issue is that it can be enabled at all, this environment variable worked for me:
> 
> ```
> GTK_THEME=Adwaita:dark
> ```
> 
> In KDE you can just add it to the applications launch configuration.

thanks. was missing the ":dark" specifier:

<img width="1236" height="1036" alt="Image" src="https://github.com/user-attachments/assets/2ff6dd9d-1704-415a-b761-bf8023abd5a1" />

**Comment by SoftFever** [comment]
The Flathub build is now live!  
Shoutout to @powpingdone and @hadess for laying the groundwork, and to everyone who helped along the way ♥️

**Comment by SoftFever** [comment]
> https://flathub.org/en/apps/com.orcaslicer.OrcaSlicer
> 
> Already available on Flathub. Don't forget to transfer your data from the old app_id to the new one. I just did `mv ~/.var/app/{io.github.softfever.OrcaSlicer,com.orcaslicer.OrcaSlicer}`.

I actually added a migration feature: OrcaSlicer can automatically detect `~/.var/app/{io.github.softfever.OrcaSlicer` and copy the configs to the new location.

**Comment by eternal-sorrow** [comment]
> I actually added a migration feature: OrcaSlicer can automatically detect `~/.var/app/{io.github.softfever.OrcaSlicer` and copy the configs to the new location.

Yes, but it uses wrong appID for some reason. `io.github.orcaslicer.OrcaSlicer` instead of `io.github.softfever.OrcaSlicer`.

**Comment by eternal-sorrow** [comment]
Also, I usually strip all unneeded filesystem access from flatpak apps before I run them for the first time.

**Comment by gioan777** [comment]
OK, orcaslicer flatpak just crashes for me on Arch Linux. After opening a debugger it seems it crashes when handling locales.
```
Executable /app/bin/orca-slicer doesn't seem to be a flatpaked application.
Running: `"flatpak" "run" "--filesystem=home" "--filesystem=/tmp" "--command=gdb" "--devel" "com.orcaslicer.OrcaSlicer" "/app/bin/orca-slicer" "/tmp/tmptx6zkgq5"`
GNU gdb (GDB) 17.1
Copyright (C) 2025 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /app/bin/orca-slicer...
(No debugging symbols found in /app/bin/orca-slicer)
[New LWP 2]

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.gnome.org/>
Enable debuginfod for this session? (y or [n]) y
Debuginfod has been enabled.
To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/app/bin/orca-slicer'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007fa01e6fd51c in __pthread_kill_implementation () from /usr/lib/x86_64-linux-gnu/libc.so.6
(gdb) bt full
#0  0x00007fa01e6fd51c in __pthread_kill_implementation () at /usr/lib/x86_64-linux-gnu/libc.so.6
#1  0x00007fa01e6a33be in raise () at /usr/lib/x86_64-linux-gnu/libc.so.6
#2  0x00007fa01e68a8ed in abort () at /usr/lib/x86_64-linux-gnu/libc.so.6
#3  0x00007fa0260b304b in __gnu_cxx::__verbose_terminate_handler() [clone .cold] () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007fa0260cc38a in __cxxabiv1::__terminate(void (*)()) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007fa0260b2a8a in std::terminate() () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007fa0260cc640 in __cxa_throw () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x00007fa0260b7047 in std::__throw_runtime_error(char const*) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#8  0x00007fa0260b53a6 in std::locale::facet::_S_create_c_locale(__locale_struct*&, char const*, __locale_struct*) [clone .cold] () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x00007fa0260fbb4d in std::locale::_Impl::_Impl(char const*, unsigned long) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#10 0x00007fa0260fcfea in std::locale::locale(char const*) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#11 0x000056074f3abd94 in ??? ()
#12 0x000056074f3abfa4 in ??? ()
#13 0x000056074d6d5366 in ??? ()
#14 0x00007fa01e68caa4 in __libc_start_main () at /usr/lib/x86_64-linux-gnu/libc.so.6
#15 0x000056074b68f365 in ??? ()
(gdb) 
```

**Comment by SoftFever** [comment]
> > I actually added a migration feature: OrcaSlicer can automatically detect `~/.var/app/{io.github.softfever.OrcaSlicer` and copy the configs to the new location.
> 
> Yes, but it uses wrong appID for some reason. `io.github.orcaslicer.OrcaSlicer` instead of `io.github.softfever.OrcaSlicer`.

Oops, I forgot that 2.3.1 was using `io.github.softfever.OrcaSlicer`. `io.github.orcaslicer.OrcaSlicer` was used briefly during the 2.3.2 dev builds.

**Comment by SoftFever** [comment]
> Also, I usually strip all unneeded filesystem access from flatpak apps before I run them for the first time.

I believe OrcaSlicer can only access files in `~/.var/app/io.github.orcaslicer.OrcaSlicer` and your home folder. Please let me know in case you noticed some excessive file access

**Comment by eternal-sorrow** [comment]
>io.github.orcaslicer.OrcaSlicer was used briefly during the 2.3.2 dev builds.

I don't use dev builds.

>Please let me know in case you noticed some excessive file access

I'm not the right person to ask this. For me any file access is excessive unless it's absolutely needed. I only allow specific directories that store files intended for this specific app.

But in general I think giving any app access to the whole home directory is a big no no.

**Comment by SoftFever** [comment]
> > io.github.orcaslicer.OrcaSlicer was used briefly during the 2.3.2 dev builds.
> 
> I don't use dev builds.
> 
> > Please let me know in case you noticed some excessive file access
> 
> I'm not the right person to ask this. For me any file access is excessive unless it's absolutely needed. I only allow specific directories that store files intended for this specific app.
> 
> But in general I think giving any app access to the whole home directory is a big no no.

I thought that was pretty common. For example, users typically load the project and models from somewhere in their home folder and save the model/project/G-code somewhere there as well.

**Comment by SoftFever** [comment]
> OK, orcaslicer flatpak just crashes for me on Arch Linux. After opening a debugger it seems it crashes when handling locales.
> 
> ```
> Executable /app/bin/orca-slicer doesn't seem to be a flatpaked application.
> Running: `"flatpak" "run" "--filesystem=home" "--filesystem=/tmp" "--command=gdb" "--devel" "com.orcaslicer.OrcaSlicer" "/app/bin/orca-slicer" "/tmp/tmptx6zkgq5"`
> GNU gdb (GDB) 17.1
> Copyright (C) 2025 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> Type "show copying" and "show warranty" for details.
> This GDB was configured as "x86_64-unknown-linux-gnu".
> Type "show configuration" for configuration details.
> For bug reporting instructions, please see:
> <https://www.gnu.org/software/gdb/bugs/>.
> Find the GDB manual and other documentation resources online at:
>     <http://www.gnu.org/software/gdb/documentation/>.
> 
> For help, type "help".
> Type "apropos word" to search for commands related to "word"...
> Reading symbols from /app/bin/orca-slicer...
> (No debugging symbols found in /app/bin/orca-slicer)
> [New LWP 2]
> 
> This GDB supports auto-downloading debuginfo from the following URLs:
>   <https://debuginfod.gnome.org/>
> Enable debuginfod for this session? (y or [n]) y
> Debuginfod has been enabled.
> To make this setting permanent, add 'set debuginfod enabled on' to .gdbinit.
> [Thread debugging using libthread_db enabled]
> Using host libthread_db library "/usr/lib/x86_64-linux-gnu/libthread_db.so.1".
> Core was generated by `/app/bin/orca-slicer'.
> Program terminated with signal SIGABRT, Aborted.
> #0  0x00007fa01e6fd51c in __pthread_kill_implementation () from /usr/lib/x86_64-linux-gnu/libc.so.6
> (gdb) bt full
> #0  0x00007fa01e6fd51c in __pthread_kill_implementation () at /usr/lib/x86_64-linux-gnu/libc.so.6
> #1  0x00007fa01e6a33be in raise () at /usr/lib/x86_64-linux-gnu/libc.so.6
> #2  0x00007fa01e68a8ed in abort () at /usr/lib/x86_64-linux-gnu/libc.so.6
> #3  0x00007fa0260b304b in __gnu_cxx::__verbose_terminate_handler() [clone .cold] () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #4  0x00007fa0260cc38a in __cxxabiv1::__terminate(void (*)()) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #5  0x00007fa0260b2a8a in std::terminate() () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #6  0x00007fa0260cc640 in __cxa_throw () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #7  0x00007fa0260b7047 in std::__throw_runtime_error(char const*) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #8  0x00007fa0260b53a6 in std::locale::facet::_S_create_c_locale(__locale_struct*&, char const*, __locale_struct*) [clone .cold] () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #9  0x00007fa0260fbb4d in std::locale::_Impl::_Impl(char const*, unsigned long) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #10 0x00007fa0260fcfea in std::locale::locale(char const*) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
> #11 0x000056074f3abd94 in ??? ()
> #12 0x000056074f3abfa4 in ??? ()
> #13 0x000056074d6d5366 in ??? ()
> #14 0x00007fa01e68caa4 in __libc_start_main () at /usr/lib/x86_64-linux-gnu/libc.so.6
> #15 0x000056074b68f365 in ??? ()
> (gdb) 
> ```

what locale setting are you use?

**Comment by mtorromeo** [comment]
> I thought that was pretty common. For example, users typically load the project and models from somewhere in their home folder and save the model/project/G-code somewhere there as well.

It is common and a sensible default for this use-case. If any user wishes to lock down access to their home, they can still do so.

**Comment by eternal-sorrow** [comment]
>I thought that was pretty common. For example, users typically load the project and models from somewhere in their home folder and save the model/project/G-code somewhere there as well.

As common as it is, it undermines the security of flatpak sandbox, making it basically useless. The problem is that XDG system does not have a designated directory for projects. For me the acceptable default configuration for apps like Orca Slicer would be (in the ideal world) an access to the "XDG projects" directory and maybe read only access to the downloads directory. And that's it.

**Comment by gioan777** [comment]
> > OK, orcaslicer flatpak just crashes for me on Arch Linux. After opening a debugger it seems it crashes when handling locales.
> > ```
> > ```
> 
> what locale setting are you use?

These are my locale settings. Not sure if Flatpak affects anything, but 2.3.2-rc2 works with no issue.
```
% locale
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=en_AU.UTF-8
LC_TIME=en_DK.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=el_GR.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT=en_AU.UTF-8
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=

```

**Comment by SoftFever** [comment]
> > > OK, orcaslicer flatpak just crashes for me on Arch Linux. After opening a debugger it seems it crashes when handling locales.
> > 
> > 
> > what locale setting are you use?
> 
> These are my locale settings. Not sure if Flatpak affects anything, but 2.3.2-rc2 works with no issue.
> 
> ```
> % locale
> LANG=en_US.UTF-8
> LC_CTYPE="en_US.UTF-8"
> LC_NUMERIC=en_AU.UTF-8
> LC_TIME=en_DK.UTF-8
> LC_COLLATE="en_US.UTF-8"
> LC_MONETARY=el_GR.UTF-8
> LC_MESSAGES="en_US.UTF-8"
> LC_PAPER="en_US.UTF-8"
> LC_NAME="en_US.UTF-8"
> LC_ADDRESS="en_US.UTF-8"
> LC_TELEPHONE="en_US.UTF-8"
> LC_MEASUREMENT=en_AU.UTF-8
> LC_IDENTIFICATION="en_US.UTF-8"
> LC_ALL=
> ```

I wonder if it's similar to the issue that this PR solves: https://github.com/OrcaSlicer/OrcaSlicer/pull/12948

**Comment by gioan777** [comment]
For the record when I call it like this
`% flatpak run --env=LC_ALL=en_US.UTF-8 com.orcaslicer.OrcaSlicer`
Orcaslicer launches without any issue

**Comment by SoftFever** [comment]
> For the record when I call it like this `% flatpak run --env=LC_ALL=en_US.UTF-8 com.orcaslicer.OrcaSlicer` Orcaslicer launches without any issue

I think it's a regression from my PR: https://github.com/OrcaSlicer/OrcaSlicer/pull/12751
it might be solved by https://github.com/OrcaSlicer/OrcaSlicer/pull/12948 though

**Comment by Jookia** [comment]
As an overdue off-topic comment: Much earlier in this thread I was fairly negative about the general process here, and I want to apologize for bringing that attitude. I should have said this much sooner, but I guess today's the day for it. I also want to thank you for your work on this feature and the whole of OrcaSlicer. (I also gave a small donation as thanks but it will never make up for the actual cost of software development unfortunately)

**Comment by neclimdul** [comment]
Awesome work! Thanks for sticking it out and getting this done everyone!

**Comment by Demindiro** [comment]
For a fresh install, with both 2.3.2 and nightly from either Github or Flathub I get the following error when trying to run OrcaSlicer:
```
flatpak run --filesystem=/tmp com.orcaslicer.OrcaSlicer
New Flatpak data dir: "/home/xxx/.var/app/com.orcaslicer.OrcaSlicer/config/OrcaSlicer"
Inconsistent value (1) for DRI_PRIME. Should be < 1 (GPU devices count). Using: 0

(orca-slicer:2): Gtk-WARNING **: 19:51:27.601: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../gtk/gtkiconhelper.c:495:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Adwaita/scalable/status/image-missing.svg: Loader process exited early with status '1'Command:
 env -i XDG_RUNTIME_DIR="/run/user/1002" "flatpak-spawn" "--sandbox" "--watch-bus" "--directory=/" "--forward-fd=31" "prlimit" "--as=17012097024" "/usr/libexec/glycin-loaders/2+/glycin-svg" "--dbus-fd" "31" (gdk-pixbuf-error-quark, 0)
Bail out! Gtk:ERROR:../gtk/gtkiconhelper.c:495:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Adwaita/scalable/status/image-missing.svg: Loader process exited early with status '1'Command:  env -i XDG_RUNTIME_DIR="/run/user/1002" "flatpak-spawn" "--sandbox" "--watch-bus" "--directory=/" "--forward-fd=31" "prlimit" "--as=17012097024" "/usr/libexec/glycin-loaders/2+/glycin-svg" "--dbus-fd" "31" (gdk-pixbuf-error-quark, 0)
```
I have tried some workarounds but none have any effect. A fresh install of 2.3.1 does work however.

OS: Guix. I tried relogging (and restarting X11) to no effect.

**Comment by Olypolyu** [comment]
```
oly@kheputer:~$ flatpak list | grep -i "orca"
oly@kheputer:~$ flatpak install orcaslicer
Looking for matches…
Found ref ‘app/com.orcaslicer.OrcaSlicer/x86_64/stable’ in remote ‘flathub’ (system).
Use this ref? [Y/n]: y

com.orcaslicer.OrcaSlicer permissions:
    ipc                           network       x11       devices       file access [1]
    system dbus access [2]

    [1] /media, /run/media, /run/spnav.sock:ro, home, xdg-run/gvfs, ~/.var/app/io.github.orcaslicer.OrcaSlicer:ro
    [2] org.freedesktop.UDisks2


        ID                                  Branch          Op          Remote           Download
 1. [✓] com.orcaslicer.OrcaSlicer           stable          i           flathub          113.8 MB / 173.6 MB

Installation complete.
oly@kheputer:~$ flatpak run com.orcaslicer.OrcaSlicer
terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
```

Well, this is on fedora 43 with KDE. Haven't tried on a fresh install. Was working fine previously before the change to flathub.

**Comment by kaizoor** [comment]
It seams that i depends on some ENV settings of the current session. I encounter the same issue as mentioned by @Olypolyu 

If I run it as follows Orca is able to start:
`export LC_ALL=C; unset LANGUAGE; flatpak run com.orcaslicer.OrcaSlicer`

**Comment by Pandapip1** [comment]
> I thought that was pretty common. For example, users typically load the project and models from somewhere in their home folder and save the model/project/G-code somewhere there as well.

This doesn't need all files access. Flatpak applications typically use [XDG desktop portal](https://wiki.archlinux.org/title/XDG_Desktop_Portal) to open a file picker and (to the best of my understanding) get a fd you can read/write to.

**Comment by Olypolyu** [comment]
@kaizoor, oh thanks.

`sudo flatpak override --env=LC_ALL=C --unset-env=LANGUAGE com.orcaslicer.OrcaSlicer`
Adapted your command a bit with the flatpak lingo. Now it is solved "permanently" for me at least. huge thanks.

**Comment by Scout339** [comment]
FINALLY it's on Flathub! Thank you guys!

**Comment by neclimdul** [comment]
Feel like since its out there now we can probably close this and address additional hiccups with new issues?
